Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs

```diff
@@ -0,0 +1,727 @@
+            }
+            else
+            {
+                column.SortingOrder = SortingOrder.None;
+                column.SortingIconContainer.IsVisible = false;
+            }
+        }
+
+        return items;
+    }
+
+    private void SortColumn(SortData? sortData = null)
+    {
+        if (ItemsSource is null)
+        {
+            return;
+        }
+
+        sortData ??= SortedColumnIndex;
+
+        var originalItems = ItemsSource.Cast<object>().ToList();
+
+        IList<object> sortedItems;
+
+        if (CanSort(sortData))
+        {
+            sortedItems = GetSortedItems(originalItems, sortData);
+        }
+        else
+        {
+            sortedItems = originalItems;
+        }
+
+
+        InternalItems = sortedItems;
+    }
+    #endregion Sorting methods
+
+    #region Methods
+
+    /// <summary>
+    /// Scrolls to the row
+    /// </summary>
+    /// <param name="item">Item to scroll</param>
+    /// <param name="position">Position of the row in screen</param>
+    /// <param name="animated">animated</param>
+    public void ScrollTo(object item, ScrollToPosition position, bool animated = true) => _collectionView.ScrollTo(item, position: position, animate: animated);
+
+    #endregion Methods
+
+    #region Bindable properties
+    public static readonly BindableProperty ReadOnlyListProperty = BindableProperty.Create(nameof(ReadOnlyList), typeof(List<int>), typeof(DataGrid), defaultValueCreator: _ => new List<int>());
+    public static readonly BindableProperty ItemSizingStrategyProperty = BindableProperty.Create(nameof(ItemSizingStrategy), typeof(ItemSizingStrategy), typeof(DataGrid), DeviceInfo.Platform == DevicePlatform.Android ? ItemSizingStrategy.MeasureAllItems : ItemSizingStrategy.MeasureFirstItem);
+    public static readonly BindableProperty ColumnsProperty = BindableProperty.Create(nameof(Columns), typeof(ObservableCollection<DataGridColumn>), typeof(DataGrid), propertyChanged: OnColumnChanged, defaultValueCreator: _ => new ObservableCollection<DataGridColumn>());
+    public static readonly BindableProperty ItemsSourceProperty = BindableProperty.Create(nameof(ItemsSource), typeof(IEnumerable), typeof(DataGrid), null, propertyChanged: OnItemsSourceChanged);
+    public static readonly BindableProperty IsSortableProperty = BindableProperty.Create(nameof(IsSortable), typeof(bool), typeof(DataGrid), false);
+    public static readonly BindableProperty SelectionModeProperty = BindableProperty.Create(nameof(SelectionMode), typeof(SelectionMode), typeof(DataGrid), Microsoft.Maui.Controls.SelectionMode.None, BindingMode.TwoWay, propertyChanged: OnSelectionModeChanged);
+    public static readonly BindableProperty SortIconProperty = BindableProperty.Create(nameof(SortIcon), typeof(Polygon), typeof(DataGrid));
+    public static readonly BindableProperty NoDataViewProperty = BindableProperty.Create(nameof(NoDataView), typeof(View), typeof(DataGrid), propertyChanged: OnNoDataViewChanged);
```

Comment:
Rename this to`EmptyView`. That's the standard MAUI uses for all other layouts

Reply:
Done